base/android/java/src/org/chromium/base/ResourceExtractor.java

```diff
@@ -114,13 +128,19 @@ private void doInBackgroundImpl() {
             try {
                 // Extract all files that don't already exist.
                 for (ResourceEntry entry : sResourcesToExtract) {
-                    File output = new File(outputDir, entry.extractedFileName);
+                    File dir = isAppDataFile(entry.extractedFileName) ? appDataDir : outputDir;
```

Comment:
I think this part of the change deserves an explanation in a comment: as I mentioned in JIRA, we only need to do this (and bring back `isAppDataFile()`) because of what we do with `icudtl.dat` (and the V8 snapshot files if we ever use them on Android): loading them from `assets/` currently does not work with either embedded mode (the file is in `raw/res`) or shared mode (the app's context is used to retrieve the `AssetManager`, not Crosswalk's).
We thus need to put those special files in a different directory so that we leverage the fallback code in Chromium to load these files from disk.